src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -372,6 +376,8 @@ private AIState lookForNeededItems()
 
     /**
      * Updates the itemsCurrentlyNeeded with current values.
+     * 
+     * This function seems bugged, as it gives itemsCurrentlyNeeded = isEmpty 
```

Comment:
steps to reproduce:
1. Place builders chest
2. Click `Build Builders hut` but don't give him any materials except the tools he needs(shovel, axe, pickaxe)
3. Let him clear the Area
4. The builder will (as intended) run to his chest and request the materials he needs
5. Dont give him all materials. for example if he requests 6 cobblestone, only give him 2

The builder then will leave "danger zone" and place the cobblestone. then he will stand around doing nothing.